src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Count.cs

```diff
@@ -20,7 +20,7 @@ public int Count(string input)
 
             int count = 0;
 
-            RunAllMatchesWithCallback(input, 0, ref count, static (ref int count, Match match) =>
+            RunAllMatchesWithCallback(input, RightToLeft ? input.Length : 0, ref count, static (ref int count, Match match) =>
```

Comment:
Good catch! I would have expected tests to catch this, but I'm glad you now added tests for it 👍